core/src/test/java/org/elasticsearch/cluster/metadata/IndexGraveyardTests.java

```diff
@@ -0,0 +1,170 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.ByteBufferStreamInput;
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.lessThan;
+
+/**
+ * Tests for the {@link IndexGraveyard} class
+ */
+public class IndexGraveyardTests extends ESTestCase {
+
+    public void testIsEquals() {
+        final IndexGraveyard graveyard1 = new IndexGraveyard();
+        assertTrue("empty graveyards are equal", graveyard1.isEquals(new IndexGraveyard(graveyard1)));
+        final IndexGraveyard graveyard2 = createRandom();
+        assertTrue("same graveyards are equal", graveyard2.isEquals(new IndexGraveyard(graveyard2)));
+    }
+
+    public void testSerialization() throws IOException {
+        final int numIterations = 5;
+        for (int i = 0; i < numIterations; i++) {
+            final IndexGraveyard graveyard = createRandom();
+            final BytesStreamOutput out = new BytesStreamOutput();
+            graveyard.writeTo(out);
+            final ByteBufferStreamInput in = new ByteBufferStreamInput(ByteBuffer.wrap(out.bytes().toBytes()));
+            assertTrue("writing to and reading from stream must produce the same index graveyard",
+                graveyard.isEquals(new IndexGraveyard(in)));
+        }
+    }
+
+    public void testXContent() throws IOException {
+        final int numIterations = 5;
+        for (int i = 0; i < numIterations; i++) {
+            final IndexGraveyard graveyard = createRandom();
+            final XContentBuilder builder = JsonXContent.contentBuilder();
+            graveyard.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            XContentParser parser = XContentType.JSON.xContent().createParser(builder.bytes());
+            parser.nextToken(); // the beginning of the parser
+            parser.nextToken(); // the START_ARRAY token
+            assertTrue("writing to and reading from x-content must produce the same index graveyard",
+                graveyard.isEquals(IndexGraveyard.fromXContent(parser)));
+        }
+    }
+
+    public void testAddTombstones() {
+        final int numIterations = 5;
+        for (int i = 0; i < numIterations; i++) {
```

Comment:
I meant if people want to make sure things are ok, the can put the test on repeat locally and run it - I do it a lot with these kind of random unit tests